src/TreeView/TreeView.js

```diff
@@ -0,0 +1,471 @@
+import React, { Component } from 'react';
+
+export class TreeCol extends Component {
+    render() {
+        const {
+            children,
+            className,
+            ...rest
+        } = this.props;
+
+        const cellClassName = classnames(
+            'fd-tree__col',
+            className
+        );
+
+        return (
+            <div
+                {...rest}
+                className={cellClassName}>
+                {children}
+            </div>
+        );
+    }
+}
+
+TreeCol.displayName = 'TreeCol';
+
+TreeCol.propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string
+};
+
+export class TreeHead extends Component {
+    render() {
+        const {
+            buttonProps,
+            children,
+            className,
+            isExpanded,
+            onExpandAll,
+            ...rest
+        } = this.props;
+
+        const headerClassName = classnames(
+            'fd-tree',
+            'fd-tree--header',
+            className
+        );
+
+        return (
+            <div {...rest} className={headerClassName}>
+                <div className='fd-tree__row fd-tree__row--header'>
+                    {
+                        React.Children.map(children, (child, index) => {
+                            const isFirstTreeCol = index === 0 && child.type && child.type.name === 'TreeCol';
+
+                            // Add control class to first TreeCol element
+                            const childClassName = classnames({
+                                'fd-tree__col--control': isFirstTreeCol
+                            });
+
+                            // Add expand button to first TreeCol element
+                            const newChildren = isFirstTreeCol ? (
+                                <div>
+                                    <button
+                                        {...buttonProps}
+                                        aria-label={isExpanded ? 'collapse all' : 'expand all'}
+                                        aria-pressed={isExpanded}
+                                        className='fd-tree__control'
+                                        onClick={onExpandAll} />
+                                    {child.props && child.props.children}
+                                </div>
+                            ) : child.props && child.props.children;
+
+                            return React.cloneElement(child, {
+                                children: newChildren,
+                                className: childClassName
+                            });
+                        })
+                    }
+                </div>
+            </div>
+        );
+    }
+}
+
+TreeHead.displayName = 'TreeHead';
+
+TreeHead.propTypes = {
+    buttonProps: PropTypes.object,
+    children: PropTypes.node,
+    className: PropTypes.string,
+    isExpanded: PropTypes.bool,
+    onExpandAll: PropTypes.func
+};
+
+TreeHead.propDescriptions = {
```

Comment:
I know that `children` is one of the defaulted prop descriptions, but the message is pretty generic.  Since these components now essentially depend on certain components being children of other components, maybe we should add a `children` prop description for each component specifying which child component(s) are expected (_e.g._ `TreeView` expects `TreeHead` and/or `Tree`, `TreeHead` expects `TreeCol`, etc.)

Reply:
Added more specific prop descriptions for `children` in 268f179